File: 02-programacion-basica/11-validacion-errores-y-excepciones/scala/ejemplos-guiados/ManejoErroresEntrada.scala
import scala.util.Try

object ManejoErroresEntrada:
  def convertirEntero(texto: String): Option[Int] =
    // Try captura el posible fallo de conversión; toOption lo expresa sin usar null.
    Try(texto.trim.toInt).toOption

  def main(args: Array[String]): Unit =
    val entrada = "20x"

    convertirEntero(entrada) match
      case Some(valor) => println(s"Valor: $valor")
      case None        => println("Entrada inválida: se esperaba un entero.")

File: 02-programacion-basica/11-validacion-errores-y-excepciones/java/ejemplos-guiados/ManejoErroresEntrada.java
public class ManejoErroresEntrada {
    static Integer convertirEntero(String texto) {
        try {
            return Integer.valueOf(texto.trim());
        } catch (NumberFormatException e) {
            // null se usa aquí solo con fines didácticos para representar conversión fallida.
            return null;
        }
    }

    public static void main(String[] args) {
        String entrada = "20x";
        Integer valor = convertirEntero(entrada);

        if (valor == null) {
            System.out.println("Entrada inválida: se esperaba un entero.");
        } else {
            System.out.println("Valor: " + valor);
        }
    }
}
